src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.h

```diff
@@ -8,9 +8,30 @@ struct ObjectKankyo;
 
 typedef void (*ObjectKankyoActionFunc)(struct ObjectKankyo*, GlobalContext*);
 
+typedef struct ObjectKankyoEffect {
+    /*  0x0 */ u8 state;
+    /*  0x4 */ Vec3f pos;   // relative to base
+    /* 0x10 */ Vec3f lastPos;
+    /* 0x1C */ Vec3f base;
+    /* 0x28 */ Vec3f dirPhase; // input to sinf for movement
```

Comment:
```suggestion
    /* 0x28 */ Vec3f dirPhase; // phase of the velocity wave
```
btw maybe `velPhase` idk